### PR TITLE
[Feature][Engine] Change the name of the rest-api interface for returning job info

### LINKED ISSUE #6812: [Feature][Engine] The name of the rest-api interface for returning job details was changed from running-job to job-info
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

**/hazelcast/rest/maps/running-job/:jobId**  The interface is return details of a job. The name of the interface is ambiguous, advised to change the name to  **/hazelcast/rest/maps/job-info/:jobId**

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hawk9821**
…b details was changed from running-job to job-info #6812

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Review by Hisoka-X — COMMENTED** [review]
Thanks @hawk9821 for create this PR! I think your point is right. But we'd better not modify the old url api, but create a new api and mark the old one as deprecated. This will ensure backward compatibility.

**Comment by hawk9821** [comment]
> Thanks @hawk9821 for create this PR! I think your point is right. But we'd better not modify the old url api, but create a new api and mark the old one as deprecated. This will ensure backward compatibility.

Thanks @Hisoka-X,   I have modified it as requested

File: docs/en/seatunnel-engine/rest-api.md [hunk 1]
```diff
@@ -77,7 +77,62 @@ network:
...
+{
+  "jobId" : ""
+}
+```
+
+</details>
+
+------------------------------------------------------------------------------------------
+
+### Return details of a job @Deprecated.
```
Comment (Hisoka-X) [inline docs/en/seatunnel-engine/rest-api.md]:
```suggestion
### Return details of a job. This API has been deprecated, please use /hazelcast/rest/maps/job-info/:jobId instead
```

File: docs/zh/seatunnel-engine/rest-api.md [hunk 2]
```diff
@@ -75,6 +75,61 @@ network:
...
+{
+  "jobId" : ""
+}
+```
+
+</details>
+
+------------------------------------------------------------------------------------------
+
+### 返回作业的详细信息(已弃用)。
```
Comment (Hisoka-X) [inline docs/zh/seatunnel-engine/rest-api.md]:
```suggestion
### 返回作业的详细信息。此API已经启用，请使用/hazelcast/rest/maps/job-info/:jobId替代
```

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java [hunk 3]
```diff
@@ -62,13 +62,8 @@
...
 
 import static com.hazelcast.internal.ascii.rest.HttpStatusCode.SC_500;
-import static org.apache.seatunnel.engine.server.rest.RestConstant.FINISHED_JOBS_INFO;
-import static org.apache.seatunnel.engine.server.rest.RestConstant.RUNNING_JOBS_URL;
-import static org.apache.seatunnel.engine.server.rest.RestConstant.RUNNING_JOB_URL;
-import static org.apache.seatunnel.engine.server.rest.RestConstant.RUNNING_THREADS;
-import static org.apache.seatunnel.engine.server.rest.RestConstant.SYSTEM_MONITORING_INFORMATION;
 
-public class RestHttpGetCommandProcessor extends HttpCommandProcessor<HttpGetCommand> {
+public static class RestHttpGetCommandProcessor extends HttpCommandProcessor<HttpGetCommand> {
```
Comment (Hisoka-X) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java]: Why change this?

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java [hunk 4]
```diff
@@ -300,6 +296,54 @@ private void handleJobInfoById(HttpGetCommand command, String uri) {
         }
     }
 
+    private void handleJobInfoByJobId(HttpGetCommand command, String uri) {
```
Comment (Hisoka-X) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java]: I think we can reuse `handleJobInfoById` method, not create new one.

**Review by Hisoka-X — COMMENTED** [review]
Thanks for update!

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java (hunk 4 above)
Comment (hawk9821) [reply seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java]: OK

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java (hunk 3 above)
Comment (hawk9821) [reply seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java]: mvn spotless:apply，missed check .  modified

File: docs/zh/seatunnel-engine/rest-api.md (hunk 2 above)
Comment (hawk9821) [reply docs/zh/seatunnel-engine/rest-api.md]: OK

File: docs/en/seatunnel-engine/rest-api.md (hunk 1 above)
Comment (hawk9821) [reply docs/en/seatunnel-engine/rest-api.md]: OK

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java [hunk 5]
```diff
@@ -252,24 +253,20 @@ private void handleJobInfoById(HttpGetCommand command, String uri) {
...
+                        .getHazelcastInstance()
+                        .getMap(Constant.IMAP_RUNNING_JOB_INFO);
+        JobInfo jobInfo = (JobInfo) jobInfoMap.get(Long.valueOf(jobId));
+        IMap<Object, Object> finishedJobStateMap =
+                this.textCommandService
+                        .getNode()
+                        .getNodeEngine()
+                        .getHazelcastInstance()
+                        .getMap(Constant.IMAP_FINISHED_JOB_STATE);
+        JobState finishedJobState = (JobState) finishedJobStateMap.get(Long.valueOf(jobId));
```
Comment (Hisoka-X) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java:269]: Any reason for change this? Seem like no difference between before.

File: docs/en/seatunnel-engine/rest-api.md [hunk 6]
```diff
@@ -77,7 +77,62 @@ network:
...
+{
+  "jobId" : ""
+}
+```
+
+</details>
+
+------------------------------------------------------------------------------------------
+
+### Return details of a job. This API has been deprecated, please use /hazelcast/rest/maps/job-info/:jobId instead
```
Comment (Hisoka-X) [inline docs/en/seatunnel-engine/rest-api.md]:
```suggestion
### Return details of a job. 
This API has been deprecated, please use /hazelcast/rest/maps/job-info/:jobId instead
```

File: docs/zh/seatunnel-engine/rest-api.md [hunk 7]
```diff
@@ -75,6 +75,61 @@ network:
...
+{
+  "jobId" : ""
+}
+```
+
+</details>
+
+------------------------------------------------------------------------------------------
+
+### 返回作业的详细信息。此API已经弃用，请使用/hazelcast/rest/maps/job-info/:jobId替代。
```
Comment (Hisoka-X) [inline docs/zh/seatunnel-engine/rest-api.md]:
```suggestion
### 返回作业的详细信息
此API已经弃用，请使用/hazelcast/rest/maps/job-info/:jobId替代。
```

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java (hunk 5 above)
Comment (hawk9821) [reply seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java:269]: In openEuler system, call /hazelcast/rest/maps/running-job/  returned to  {} ，remote debug finds that the JobState object is not returned correctly. After the code is modified, the JobState object can be returned correctly

image
image
888f2aaf3178c2e301fc309b0f47aec

Comment (Hisoka-X) [reply seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java:269]: Feeling a little strange. cc @hailin0 @EricJoy2048

**Review by Hisoka-X — APPROVED** [review]
Anyway, it will not affect result of now.

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**